libraries/Wire/examples/master_check/master_check.ino

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2016 Intel Corporation.  All rights reserved.
+ * See the bottom of this file for the license terms.
+ */
+
+/*
+ *  This sketch is to demonstrate I2C reading and writing, to a device, as a bus master.  There 4 things it
+ *  does:  Firstly, it generates a pattern. Secondly, it writes the pattern (data) to the device.  Thirdly, it reads
+ *  the data back.  Finally, it compares between the read and write data to look for differences.  There should not
+ *  be any differences.  Otherwise, it may point to a faulty I2C device or a bad connection between the board
+ *  and the device (please check to see if the board and the device share a common ground).
+ *  To make things interesting, we make the board doing these things continuously.  In addition, we change the
+ *  pattern a bit in each lap (or loop, as they said in the computing world) so that you know where an error might
+ *  occur in the testing.  Not that we think the test will always result in some error detection.  Just in
+ *  case it does, you can tell how many laps the program run before an error occurs.
+ *
+ */
+
+#include <Wire.h>
+#define BUFFER_SIZE 8
+
+static int count = 0;  // recode the higher 4 bits of data
+static uint8_t buffer_sender[BUFFER_SIZE]; // data source for master writer
+static uint8_t buffer_receiver[BUFFER_SIZE]; // data distination for master reader
+
+void setup() 
+{
+  Wire.begin();         // join i2c bus (address optional for master)
+  Serial.begin(115200); // start serial for output
+  while (Serial)
+    ;
+}
+
+void loop() 
+{
+  count++;
+  Wire.beginTransmission(8); // transmit to device #8
+
+  // Step 1:  Making the pattern.  Note that the count is in the upper 4 bit of the byte.
+  // Step 2:  And write it out to the device too.
+  for (int i = 0; i < BUFFER_SIZE; i++) 
+  {
+    buffer_sender[i] = ((count & 0xf) << 4) | i;
```

Comment:
Please keep the examples simple-- many Arduino users would be baffled by this line.
You can use the existing Wire example sketches as a reference, you will see they are very short and simple.